File: app/src/main/kotlin/br/com/sailboat/template/ui/base/BasePresenter.java
package br.com.sailboat.template.ui.base;

import org.jetbrains.annotations.Nullable;

public abstract class BasePresenter<V extends BaseMvpContract.View> implements BaseMvpContract.Presenter {

    private V view;
    private boolean firstSession = true;

    @Override
    public void onViewCreated() {
        if (firstSession) {
            create();
            firstSession = false;
        } else {
            restart();
        }
    }

    @Override
    public void onDestroyView() {
        setView(null);
    }

    protected void create() {}
    protected void restart() {}

    @Override
    public void postResult() {}

    @Nullable
    public V getView() {
        return view;
    }

    @Override
    public void setView(@Nullable BaseMvpContract.View view) {
        this.view = (V) view;
    }
}

File: app/src/main/kotlin/br/com/sailboat/template/ui/base/mvp/BasePresenter.kt
package br.com.sailboat.template.ui.base.mvp

abstract class BasePresenterKotlin<V : BaseMvpContract.View> : BaseMvpContract.Presenter {

    private var firstSession = true
    var view: V? = null

    override fun onViewCreated() {
        if (firstSession) {
            create()
            firstSession = false
        } else {
            restart()
        }
    }

    override fun attachView(view: BaseMvpContract.View?) {
        this.view = view as V
    }

    override fun onDestroyView() {
        view = null
    }

    open fun create() {}
    open fun restart() {}

    override fun postResult() {}




}